src/Operation/CreateEncryptedCollection.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/*
+ * Copyright 2023-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Operation;
+
+// phpcs:disable SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse
+use MongoDB\BSON\Binary;
+// phpcs:enable
+use MongoDB\Driver\ClientEncryption;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Driver\Server;
+use MongoDB\Exception\InvalidArgumentException;
+
+use function array_key_exists;
+use function is_array;
+use function is_object;
+
+/**
+ * Create an encrypted collection.
+ *
+ * The "encryptedFields" option is required.
+ *
+ * This operation additionally creates related metadata collections and an index
+ * on the encrypted collection.
+ *
+ * @internal
+ * @see \MongoDB\Database::createCollection()
+ * @see \MongoDB\Database::createEncryptedCollection()
+ * @see https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#create-collection-helper
+ * @see https://www.mongodb.com/docs/manual/core/queryable-encryption/fundamentals/manage-collections/
+ */
+class CreateEncryptedCollection implements Executable
+{
+    /** @var CreateCollection */
+    private $createCollection;
+
+    /** @var CreateCollection[] */
+    private $createMetadataCollections;
+
+    /** @var CreateIndexes */
+    private $createSafeContentIndex;
+
+    /** @var string */
+    private $databaseName;
+
+    /** @var string */
+    private $collectionName;
+
+    /** @var array */
+    private $options;
+
+    /**
+     * @see CreateCollection::__construct() for supported options
```

Comment:
I can't tell you how great it felt to write an operation constructor doc block with option documentation copypasta.